packages/protocol/contracts/stability/SortedOracles.sol

```diff
@@ -160,9 +189,55 @@ contract SortedOracles is ISortedOracles, Ownable, Initializable {
       address(0)
     );
     emit OracleReported(token, msg.sender, now, value, DENOMINATOR);
+    recomputeRate(token);
+  }
+
+  function fracMulExp(FixidityLib.Fraction memory a, FixidityLib.Fraction memory b, uint256 exp)
+    internal
+    view
+    returns (FixidityLib.Fraction memory)
+  {
+    uint256 numerator;
+    uint256 denominator;
+    (numerator, denominator) = fractionMulExp(
+      a.unwrap(),
+      FixidityLib.fixed1().unwrap(),
+      b.unwrap(),
+      FixidityLib.fixed1().unwrap(),
+      exp,
+      18
+    );
+    return FixidityLib.newFixedFraction(numerator, denominator);
+  }
+
+  function recomputeRate(address token) internal {
+    uint256 originalMedian = limitedMedianRate[token];
     uint256 newMedian = rates[token].getMedianValue();
+    if (limitedMedianRateTimestamp[token] != 0 && maxMedianChangeRatePerSecond.unwrap() != 0) {
+      uint256 td = now.sub(limitedMedianRateTimestamp[token]);
+      if (td > 1 days) td = 1 days;
+      uint256 maxChange = fracMulExp(
+        FixidityLib.fixed1(),
+        maxMedianChangeRatePerSecond.add(FixidityLib.fixed1()),
+        td
```

Comment:
This doesn't feel safe to do as the exponents could grow very large...

Reply:
So need to add something to make sure it won't overflow?

Reply:
Added some checks, now it shouldn't be able to overflow.